api/v1alpha1/attester_types.go

```diff
@@ -64,6 +61,10 @@ type AttesterStatus struct {
 	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
+	// PgpSecret defines the name of the secret to use for signing. If the secret doesn't already exist it will be created.
+	// +optional
+	PgpSecret string `json:"pgpSecret"`
```

Comment:
this isn't making sense to me... by moving this field to the `status`, users are now unable to provide their own PGP keys.

I'm actually not even sure I fully understand the problem this story is trying to solve now.  if `.spec.pgpSecret` is a required field, then it makes sense that the secret needs to already exist, or one will be created.  so I'm not seeing how anything within `.spec` is being updated by the controller.

Reply:
Updated approach makes `spec.pgpSecret` a required field for the Attestor resource. Now rather than checking to see if the field has been set, we can assume the resource was created with a value for this field. If the secret exists, then the Attestor controller will obtain the secret. Else, it will generate a new signer key and create a new secret named after the provided `pgbSecret` value.